x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java

```diff
@@ -664,6 +664,14 @@ public void testSearchableSnapshotsInHotPhasePinnedToHotNodes() throws Exception
         Map<String, Object> hotIndexSettings = getIndexSettingsAsMap(restoredIndex);
         // searchable snapshots mounted in the hot phase should be pinned to hot nodes
         assertThat(hotIndexSettings.get(DataTier.TIER_PREFERENCE), is("data_hot"));
+
+        // Need to stop ILM to prevent a race between rollovers and deletion of searchable snapshots
+        assertOK(client().performRequest(new Request("POST", "_ilm/stop")));
+        try {
+            cleanUpCluster();
```

Comment:
Can't we just delete the ILM policy after the test is done instead?

Reply:
Unfortunately, we can't delete an ILM policy until we delete its backing indices first :(

Reply:
Ah right, I forgot that. I'm not confortable in calling `cleanUpCluster()` here as I think it tear down methods like this should be called once per test. 

I'm surprised we can't just delete the resources created by the test, like the datastream and the mounted index, and move on with the regular @After test logic. If the datastream is gone there should be no rolling over afterwards?

Reply:
Thank you @tlrx!  I've created #90075 to explicitely delete the data steam and the ILM policy which seems to work :+1: